contracts/BSCValidatorSet.sol

```diff
@@ -400,6 +433,52 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     // averageDistribute*rest may less than income, but it is ok, the dust income will go to system reward eventually.
   }
 
+  /*********************** For Temporary Maintenance **************************/
+  function canEnterMaintenance(address validator, uint256 index) public view returns (bool can, uint256 exitMaintenanceReward) {
+    Validator memory validatorInfo = currentValidatorSet[index];
+
+    if (
+      (MaxNumOfMaintaining == 0 || MaxMaintainingTime == 0)         // - 1. check if not start
+      || maintainingValidatorSet.length >= MaxNumOfMaintaining      // - 2. check if exceeded upper limit
+      || validatorInfo.jailed                                       // - 3. check if jailed
+      || maintainInfoMap[validator].isMaintaining                   // - 4. check if maintaining
+      || maintainInfoMap[validator].maintainStartAt > 0             // - 5. check if has Maintained
+      || currentValidatorSet.length <= 1                            // - 6. check num of remaining currentValidators
+    ) {
+      return (false, 0);
+    }
+
+    can = true;
+    exitMaintenanceReward = validatorInfo.incoming.mul(EXIT_MAINTENANCE_REWARD_RATIO).div(EXIT_MAINTENANCE_REWARD_RATIO_SCALE);
+  }
+
+  function enterMaintenance() external {
+    uint256 index = currentValidatorSetMap[msg.sender];
+    require(index > 0, "only current validators can enter maintenance");
+
+    // the actually index
+    index = index - 1;
+    (bool can, uint256 exitMaintenanceReward) = canEnterMaintenance(msg.sender, index);
+    require(can, "can not enter Temporary Maintenance");
+    _enterMaintenance(msg.sender, index, exitMaintenanceReward);
```

Comment:
we can not remove it from currentValidatorSet. If we do, when the validatorSet rotation happen at UTC 00, we wont get the message about the income of this validator.

Reply:
income of validators will recorded in `maintainingValidatorSet` while validators enter `Temporary Maintenance`